pkgs/development/tools/java/jprofiler/default.nix

```diff
@@ -0,0 +1,60 @@
+{ stdenv, lib, fetchurl, makeWrapper, makeDesktopItem, jdk }:
+
+stdenv.mkDerivation rec {
+  version = "13.0.2";
+  pname = "jprofiler";
+
+  src = fetchurl {
+    url = "https://download-gcdn.ej-technologies.com/jprofiler/jprofiler_linux_${builtins.replaceStrings ["."] ["_"]  version}.tar.gz";
+    sha256 = "sha256-x9I7l2ctquCqUymtlQpFXE6+u0Yg773qE6MvAxvCaEE=";
+  };
+
+  srcIcon = fetchurl {
+    url = "https://www.ej-technologies.com/assets/content/header-product-jprofiler@2x-24bc4d84bd2a4eb641a5c8531758ff7c.png";
+    sha256 = "sha256-XUmuqhnNv7mZ3Gb4A0HLSlfiJd5xbCExVsw3hmXHeVE=";
+  };
+
+  desktopItem = makeDesktopItem {
+      name = pname;
+      exec = pname;
+      icon = pname;
+      comment = meta.description;
+      desktopName = "JProfiler";
+      genericName = "Java Profiler Tool";
+      categories = [ "Development" ];
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    cp -r . $out
+
+    wrapProgram $out/bin/jpanalyze --set JAVA_HOME "${jdk.home}"
+    wrapProgram $out/bin/jprofiler --set JAVA_HOME "${jdk.home}"
+    wrapProgram $out/bin/jpcompare --set JAVA_HOME "${jdk.home}"
+    wrapProgram $out/bin/jpcontroller --set JAVA_HOME "${jdk.home}"
+    wrapProgram $out/bin/jpenable --set JAVA_HOME "${jdk.home}"
+    wrapProgram $out/bin/jpexport --set JAVA_HOME "${jdk.home}"
+    wrapProgram $out/bin/jpintegrate --set JAVA_HOME "${jdk.home}"
+    wrapProgram $out/bin/jpenable --set JAVA_HOME "${jdk.home}"
+
+    install -Dm644 "${desktopItem}/share/applications/"* \
+        -t $out/share/applications/
+    install -Dm644 "${srcIcon}" \
+        "$out/share/icons/hicolor/scalable/apps/jprofiler.png"
+
+    rm -f $out/bin/updater
+  '';
+
+  meta = with lib; {
+    description = "JProfiler's intuitive UI helps you resolve performance bottlenecks";
+    longDescription = ''
+      JProfiler's intuitive UI helps you resolve performance bottlenecks,
+      pin down memory leaks and understand threading issues.
+    '';
+    homepage = "https://www.ej-technologies.com/products/jprofiler/overview.html";
+    license = licenses.unfree;
+    platforms = platforms.linux;
```

Comment:
Given that it is written in Java, could this also work for other platforms?

Reply:
@azahi it supports different platforms but via different distributives which includes some different pre-build library which is used via JNA/JNI interface.

It is possible to install from linux's `.tar.gz` to macOS for example but some feature might not work.

To be honest I never tried.

Reply:
I see. I think you can leave the platform as-is or we can just ask someone to test it out on MacOS before merging.

Reply:
@azahi I've tested it on macOS ;) I'm running NixOS inside Parallels and using macOS as host system.